prusti-common/src/vir/optimizations/purification/mod.rs

```diff
@@ -151,26 +152,23 @@ impl<'a> StmtFolder for Purifier<'a> {
     fn fold_assign(&mut self, target: Expr, mut source: Expr, kind: AssignKind) -> Stmt {
         if let Expr::Local(local, _) = &target {
             if self.targets.contains(&local.name) {
-                match source.get_type() {
-                    Type::TypedRef(name) => {
-                        match name.as_str() {
-                            "bool" => {
-                                source = source.field(Field {
-                                    name: "val_bool".into(),
-                                    typ: Type::Bool,
-                                });
-                            }
-                            "i32" | "usize" | "u32" => {
-                                source = source.field(Field {
-                                    name: "val_int".into(),
-                                    typ: Type::Int,
-                                });
-                            }
-                            x => unreachable!("{}", x),
+                force_matches!(source.get_type(), Type::TypedRef(name) => {
```

Comment:
I would use the `let source_name = force_matches!(..); match source_name.as_str() { .. }` to avoid adding the indentation level.